code/modules/projectiles/guns/magic.dm

```diff
@@ -37,10 +36,14 @@
 	return charges
 
 /obj/item/gun/magic/newshot(params)
-	if(charges && chambered)
+	if(charges && chambered && !chambered.BB)
 		chambered.newshot(params)
 	return
 
+/obj/item/gun/magic/process_fire()
+	newshot()
+	return ..()
```

Comment:
Use . = ..() because it looks bettet

Reply:
newshot needs to happen before the call to super due to the weirder points of how the magic weapons works.

Reply:
Ughhhhh